control-plane/api-gateway/controllers/gateway_class_config_controller.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	gatewayClassConfigFinalizer = "gateway-class-exists-finalizer.consul.hashicorp.com"
+)
+
+type Controller struct {
```

Comment:
Can we rename this to `GatewayClassConfigController`? This will be in the same namespace as all the other controllers and they'll need to be named differently to avoid collisions.

Additionally, a short comment above the struct would be great to explain what this controller's responsibility is.

Reply:
done